### PR TITLE
ddimtirov - supporting hash collisions, should have fixed #101

### LINKED ISSUE #101: Fix tests (ddimtirov)
Hey @ddimtirov, since your submission we've added a basic test suite for covering some corner cases (as we discussed already). One of the tests fails for your submission. Could you take a look and fix this so the submission can be re-evaluated? Thx!

```
./test.sh ddimtirov
Validating calculate_average_ddimtirov.sh -- src/test/resources/samples/measurements-1.txt
PT0.029701S

real	0m0.337s
user	0m0.093s
sys	0m0.130s
1c1
< {=0.0/19.8/19.8, Kunming=0.0/19.8/19.8, g=0.0/19.8/19.8, ing=0.0/19.8/19.8, ming=0.0/19.8/19.8, ng=0.0/19.8/19.8, nming=0.0/19.8/19.8}
---
> {Kunming=19.8/19.8/19.8}
```

### PR DESCRIPTION
**Pull request by ddimtirov**
#### Check List:
- [x] Tests pass (`./test.sh <username>`)
- [x] All formatting changes by the build are committed
- [x] Output matches that of calculate_average_baseline.sh

* Execution time:  3.7741536
* Execution time of reference implementation: 135.6793351

### CODE REVIEW

**Comment by ddimtirov** [comment]
I bundled a `.gitattrs` file that explicitly sets line ending for Unix files - makes life easier if you develop on Windows.

**Comment by gunnarmorling** [comment]
00:15.956, nice. Thx for the update, appreciate it!

> I bundled a .gitattrs file that explicitly sets line ending for Unix files - makes life easier if you develop on Windows.

That's great, someone else just had that issue too. Should have had that from the get-go. Thx!

File: calculate_average_ddimtirov.sh [hunk 1]
```diff
@@ -17,5 +17,7 @@
 
 # --enable-preview to use the new memory mapped segments
 # We don't allocate much, so just give it 1G heap and turn off GC; the AlwaysPreTouch was suggested by the ergonomics
-JAVA_OPTS="--enable-preview -Xms1g -Xmx1g -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC  -XX:+AlwaysPreTouch"
+JAVA_OPTS="--enable-preview -da -dsa -Xms1g -Xmx1g -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC  -XX:+AlwaysPreTouch"
```
Comment (AlexanderYastrebov) [inline calculate_average_ddimtirov.sh:20]: Please add ` -XX:-EnableJVMCI` t omit jvm warning 
```
[0.001s][warning][gc,jvmci] Setting EnableJVMCI to false as selected GC does not support JVMCI: epsilon gc
```

Comment (ddimtirov) [reply calculate_average_ddimtirov.sh:20]: This is likely because the wrong VM has been used (I assume Temurin, not GraalVM).
My new PR should fix that.
